lib/active_merchant/billing/gateways/banwire.rb

```diff
@@ -32,61 +54,116 @@ def add_response_type(post)
+        post[:SHIP_MTHD_CD] = address[:method] if address[:method] #- Shipping method (optional) [text - lenght: 1] [C=low cost, D=own shipping method, I=international, M=military, N=next day, O=others, P=store pickup, T=2 day shipping, W=3 day shipping] (Example: D)
+        post[:SHIP_ADDR1] = address[:address] if address[:address] #- Shipping address (optional) [alphanumeric - lenght: 1-30] (Example: La esperanza 14)
+        post[:SHIP_CITY] = address[:city] if address[:city] #- City shipping address (optional) [text - lenght: 1-20] (Example: Mexico)
+        post[:SHIP_STPR_CD] = address[:state] if address[:state] #- State shipping address (optional) [text - lenght: 2] (ISO 3166-2) (Example: DF)
+        post[:SHIP_POSTAL_CD] = address[:zip] if address[:zip] #- Zip shipping address (optional) [alphanumeric - lenght: 1-10] (Example: 54321)
+        post[:SHIP_CNTRY_CD] = address[:contry] if address[:contry] #- Country shipping address (optional) [text - lenght: 3] (ISO 3166-1) (Example: MX)
       end
 
-      def card_brand(card)
-        brand = super
-        ({"master" => "mastercard", "american_express" => "amex"}[brand] || brand)
+      def add_address(post, options)
+        address = (options[:billing_address] || {})
+
+        post[:CARD_AVS_ADDR] = address[:address] if address[:address] #- Billing address (required) [alphanumeric - lenght: 1-30] (Example: La esperanza 14)
+        post[:CARD_AVS_ZIPCODE] = address[:zip] if address[:zip] #- Zip billing address (required) [alphanumeric - lenght: 5-7] (Example: 54321)
+      end
+
+      def add_payment(post, payment, options)
+        post[:CARD_NUM] = payment.number #- Card number (required) [numeric - lenght: 15-16] (Example: 5134422031476272)
+        post[:CARD_OWN] = payment.name #- Card owner name (required) [text - lenght: 1-30] (Example: Carlos Garcia Valles)
+        post[:CARD_EXP_DT] = sprintf("%02d", payment.month)+payment.year.to_s[-2,2] #- Card expiration date (required) [text - lenght: 4] [Format:MMYY] (Example: 1219)
+        post[:CARD_CVV] = payment.verification_value #- Card CVV (required) [text - lenght: 3-4] (Example: 162)
+        post[:CARD_TYPE] = card_brand(payment) #- Card type (required) [text - lenght:4-10] (Example: MASTERCARD)
+
+        add_address(post, options)
       end
 
       def parse(body)
+        return {} unless body
         JSON.parse(body)
       end
 
-      def commit(money, parameters)
-        raw_response = ssl_post(URL, post_data(parameters))
+      def commit(url_action, parameters)
+        success = false
+        message = ""
+        url = (test? ? test_url : live_url) + url_action
+        raw_response = ssl_post(url, post_data(parameters))
         begin
           response = parse(raw_response)
+          success = true if ((response["ID"] && response["AUTH_CODE"]) || (response["folio"] && (response["status"] == "ok")))
```

Comment:
@wvanbergen 

success = true if (response["ID"] && response["AUTH_CODE"]) || (response["folio"] && response["status"] == "ok")

is it correct?